blockonomics-woocommerce.php

```diff
@@ -408,23 +408,42 @@ function show_basic() {
     }
     function bnomics_display_tx_info($order, $email=false)
     {
-        $blockonomics = new Blockonomics();
-        $active_cryptos = $blockonomics->getActiveCurrencies();
-        foreach ($active_cryptos as $crypto) {
-            $txid = get_post_meta($order->get_id(), 'blockonomics_'.$crypto['code'].'_txid', true);
-            $address = get_post_meta($order->get_id(), $crypto['code'].'_address', true);
-            if ($txid && $address) {
-                if ($crypto['code'] == 'btc') {
-                    $base_url = Blockonomics::BASE_URL;
-                }else{
-                    $base_url = Blockonomics::BCH_BASE_URL;
-                }
-                echo '<b>'.__('Payment Details', 'blockonomics-bitcoin-payments').'</b><p><strong>'.__('Transaction', 'blockonomics-bitcoin-payments').':</strong>  <a href =\''. $base_url ."/api/tx?txid=$txid&addr=$address'>".substr($txid, 0, 10). '</a></p>';
-                if (!$email) {
-                   echo '<p>'.__('Your order will be processed on confirmation of above transaction by the bitcoin network.', 'blockonomics-bitcoin-payments').'</p>';
-                } 
+        $txids = get_post_meta($order->get_id(), 'blockonomics_payments_txids', true);
+        $addresses = get_post_meta($order->get_id(), 'blockonomics_payments_addresses', true);
+
+        if (!$txids || !$addresses) {
+            return;
+        }
+
+        $txidArray = explode(", ", $txids);
+        $addressArray = explode(", ", $addresses);
+        $txidCount = count($txidArray);
+        $bchStartLetters = ['q', 'p'];
+
+        echo '<b>'.__('Payment Details', 'blockonomics-bitcoin-payments').'</b><p><strong>'.__('Transaction', 'blockonomics-bitcoin-payments').': </strong>';
+                    
+        for ($i = 0; $i < $txidCount; $i++) {
```

Comment:
Using the txids and addresses from post meta does have some rare cases which will lead to incorrect transaction links. If the user:

- Generates btc address, but does not pay.
- Generates bch address, and pays to this address.

It may be better to fetch the transactions from the blockonomics_payments database table. This will also include the crypto for the address, and will eliminate checking the start letters to know which crypto. Each txid will have the correct corresponding address and crypto.

Reply:
Good Idea !

Reply:
Order invoice in mail view:![image](https://user-images.githubusercontent.com/91294460/234806333-45243d84-bd91-48f3-b096-dea7311eaeed.png)

Order invoice on the website :
![image](https://user-images.githubusercontent.com/91294460/234806807-bb7b3a9a-f85d-4472-9dc1-0c4c77192bab.png)

The logic for transaction fetching from the blockonomics_payments database table is done.